pkg/controller/node/nodecontroller.go

```diff
@@ -575,15 +600,29 @@ func (nc *NodeController) monitorNodeStatus() error {
 		}
 	}
 	nc.handleDisruption(zoneToNodeConditions, nodes)
+	nc.updateEvictionMetric(Evictions10Minutes, nc.evictions10Minutes)
+	nc.updateEvictionMetric(Evictions1Hour, nc.evictions1Hour)
 
 	return nil
 }
 
+func (nc *NodeController) updateEvictionMetric(metric *prometheus.GaugeVec, data *evictionData) {
+	data.slideWindow()
+	zones := data.getZones()
+	metric.Reset()
```

Comment:
This is kind of risky. If someone will hit /metrics right after this is called, she may get 0, even though in fact there aren't really 0 there.
Is it possible to do it in a different way?